### DISCUSSION TITLE
Memory Leak and 100% CPU Usage Issue on GitHub Page

### DISCUSSION THREAD

**Discussion by createOne999** [community/community#144489] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Bug

### Body

Hello,

I’ve encountered an issue on a specific GitHub page involving a memory leak and 100% CPU usage (1 core). This issue occurs across multiple browsers, including Firefox, Chrome, and Microsoft Edge. I’ve attached screenshots showing the memory and CPU usage spikes.

Environment:

Browsers: Firefox, Chrome, Microsoft Edge
Reproduction Steps: https://github.com/langchain-ai/langchain/compare/langchain%3D%3D0.2.5...langchain%3D%3D0.3.7
Issue Details:

Upon visiting this page, memory usage continues to increase, and CPU usage spikes to 100% on one core.
Over time, the page becomes progressively slower, and in some cases, it even causes the browser to crash.

![is_memory_leak](https://github.com/user-attachments/assets/532dfe8e-b430-42ec-8c1d-b2edce0574f1)

I’m just reporting this issue for GitHub’s awareness—no need for follow-up.

Thank you.

**Comment by vlovini** [comment]
Facing the same issue with Microsoft Edge.

**Comment by Invertex** [comment]
Just ran into this as well, one this page: https://github.com/skyfeed-dev/query-engine/blob/bc1601de6cdbc320bfb05a20af8e5f7f94ac602d/src/main.rs#L2596
<img width="1174" height="119" alt="image" src="https://github.com/user-attachments/assets/20d8ceb3-c2ba-445f-9ee1-24bf2353458f" />

**Comment by TheSecurityDev** [comment]
I had a row of Github repos open that were taking up ~900mb EACH compared to ~200-300mb for most other pages.

**Comment by maddada** [comment]
Keeps happening to me too. 
If I leave a PR page open then it slows my laptop to a crawl after some time.

**Comment by hunterkepley** [comment]
still happening in the big 2026, using Zen (mozilla-based)

I have to restart my browser every couple of hours while working

**Comment by jazeee** [comment]
Chrome on Kubuntu, dark mode. github takes up a couple of CPU, even if the tab is hidden.
On a whim, I tested in Incognito, and don't see an issue - which led me to looking at my extensions.
(On this current discussion, the CPU was high, so I removed the official React dev tool extension and CPU immediately cleared up)

In any case, I'd test incognito first, to eliminate extensions as the culprit.
